### PR TITLE
Updating _resolve_numeric_only function of GroupBy

### LINKED ISSUE #42395: BUG: DataFrame.groupby drops timedelta column in v1.3.0
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
df = pd.DataFrame(
    {
        "duration":[pd.Timedelta(i, unit="days") for i in range(1,6)],
        "value":[1,0,1,2,1]
    }
)

df.groupby("value").sum()
```
results in:
```
Empty DataFrame
Columns: []
Index: [0, 1, 2]

```
The same behaviour occurs whether it's sum, mean, median etc

Note the following gives close to the expected result:
```python
df.groupby("value")["duration"].sum()
```

```
value
0   2 days
1   9 days
2   4 days
Name: duration, dtype: timedelta64[ns]
```

Also, this problem may be unique to Timedeltas as the following similar example, has no problem:
```python
df = pd.DataFrame(
    {
        "duration":[i for i in range(1,6)],
        "value":[1,0,1,2,1]
    }
)

df.groupby("value").sum()
```

gives:

```
       duration
value
0             2
1             9
2             4
```

#### Problem description
In v1.3.0 (and master, commit dad3e7) when performing a groupby operation on a dataframe a timedelta column goes missing.
This behaviour does not occur in pandas 1.2.5

#### Expected Output
The expected output is given by pandas 1.2.5:
```python
      duration
value
0       2 days
1       9 days
2       4 days
```

It is a dataframe, with one column "duration", indexed by the groupby keys

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : f00ed8f47020034e752baf0250483053340971b0
python           : 3.7.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.3.0
numpy            : 1.21.0
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 19.2.3
setuptools       : 41.2.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### LINKED ISSUE #43108: BUG: Regression from 1.2.5 to 1.3.x:  groupby using sum on DataFrame containing lists fails
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

Version 1.3.2:

```python
>>> df=pd.DataFrame([["M", [1]], ["M", [2]], ["W", [10]], ["W", [20]]], columns=["MW", "v"])
>>> df.groupby("MW").sum()
Empty DataFrame
Columns: []
Index: [M, W]
>>> df.groupby("MW").v.sum()
MW
M      [1, 2]
W    [10, 20]
Name: v, dtype: object
```

Version 1.2.5:
```python
>>> df=pd.DataFrame([["M", [1]], ["M", [2]], ["W", [10]], ["W", [20]]], columns=["MW", "v"])
>>> df.groupby("MW").sum()
           v
MW
M     [1, 2]
W   [10, 20]
>>> df.groupby("MW").v.sum()
MW
M      [1, 2]
W    [10, 20]
Name: v, dtype: object
```
#### Problem description

With 1.2.5 and earlier, if the elements of one of the columns are objects that are lists, then the `groupby` operation with `sum` creates the aggregation of the lists.  With 1.3.2, it doesn't.  

#### Expected Output

Same as 1.2.5

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 3af1a4fa27a7d48d8e6aea66bf2da6fdf2e1550e
python           : 3.8.8.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19041
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_United States.1252

pandas           : 1.4.0.dev0+475.g3af1a4fa27
numpy            : 1.21.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.1.3
setuptools       : 49.6.0.post20210108
Cython           : 0.29.23
pytest           : 6.2.4
hypothesis       : 6.14.1
sphinx           : 3.5.4
blosc            : None
feather          : None
xlsxwriter       : 1.4.3
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.3
IPython          : 7.25.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 2021.05.0
fastparquet      : 0.6.3
gcsfs            : 2021.05.0
matplotlib       : 3.3.3
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 2.0.0
pyxlsb           : None
s3fs             : 0.4.2
scipy            : 1.7.0
sqlalchemy       : 1.4.20
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.18.2
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.52.0

</details>

### PR DESCRIPTION
**Pull request by kurchi1205**
- [x] closes #43108 and #42395
   It  sets the numeric_only value based on the columns on which aggregate functions are applied
- [x] tests added / passed
   New Test is added to  `test_groupby_aggregation_mixed_dtype` function
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry
   Checking the column DataTypes in `_resolve_numeric_only` function 

I am absolutely new to contributing on Open Source . So please guide me through it .

### CODE REVIEW

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (jreback) [inline pandas/tests/groupby/aggregate/test_aggregate.py:116]: make a new test

File: pandas/core/groupby/groupby.py
Comment (jreback) [inline pandas/core/groupby/groupby.py:1123]: this is very complicated, see if you can do this in a simpler way

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py
Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1123]: ok , working on it

Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1123]: I have updated a simpler version

File: pandas/core/groupby/groupby.py
Comment (jbrockmendel) [inline pandas/core/groupby/groupby.py:13]: this is already imported below

File: pandas/core/groupby/groupby.py
Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py:1123]: ill need to look at the linked issue more closely, but im very skeptical that this method is the right place for a fix.

Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1123]: Explicitly numeric_only =False solves the error , so I looked into `_resolve_numeric_only` function.
It only checked for dataframe groupby object and set numeric_only to True , which forced the non_numeric columns out of the aggregation . That's why I included checks for multi datatype  dataframe .

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (kurchi1205) [reply pandas/tests/groupby/aggregate/test_aggregate.py:116]: I will work on that tomorrow

File: pandas/core/groupby/groupby.py
Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py:1123]: The _point_ of `numeric_only` is to force non-numeric columns out of the aggregation.  Changing the behavior here "fixed" the case you tested, but also changed a bunch of other behavior that broke a bunch of other tests

**Comment by jbrockmendel** [comment]
Commented on #43108, I think this is correct as-is.

File: pandas/core/groupby/groupby.py
Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1123]: OK I get ur part, I will check for numeric columns, if there are none only then it should set numeric_only to false, that should solve the issue and not violate the documentation

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (kurchi1205) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: I have developed a new test case .

File: pandas/core/groupby/groupby.py
Comment (jbrockmendel) [reply pandas/core/groupby/groupby.py]: I'm not sure you do.

`numeric_only` is an argument passed by the user, or defaulting to True if not explicitly passed.  We should _never_ be changing it based on what dtypes are actually present.

Comment (kurchi1205) [reply pandas/core/groupby/groupby.py]: But why can't the default value be set checking the datatypes , why does it have to be True always .

**Comment by jbrockmendel** [comment]
Changing the default is something we can do, but that is an API change, not a bugfix, and would require a deprecation cycle.

**Comment by kurchi1205** [comment]
ok , so how to go about with that ? I don't know how that works ?

**Comment by jbrockmendel** [comment]
> ok , so how to go about with that ? I don't know how that works ?

First there needs to be consensus that we _want_ to change the behavior.  That discussion seems to be in #43108.  Then a PR that doesn't change the behavior, but does issue a warning that the behavior will change in a future version, e.g. #42738

**Comment by kurchi1205** [comment]
ok So I will create a different pull request and issue a warning in groupby.py

**Comment by simonjayhawkins** [comment]
@kurchi1205 I've closed #43108 as a duplicate of #42395.

but okay to have tests for both issue reports. can you also add the code sample from #42395

**Comment by kurchi1205** [comment]
ok I will add the test for issue #42395

**Comment by kurchi1205** [comment]
@simonjayhawkins  ok So I will create a different pull request and issue a warning in groupby.py 
and write the code for changing the behavior in the current pull request . 
I am sort of stuck in this deprecation cycle process

**Comment by simonjayhawkins** [comment]
> So I will create a different pull request and issue a warning in groupby.py

no need to open another PR, can keeping pushing to this one as the issue discussion progresses.

we don't deprecate behavior in a patch release so if this is the way forward, this PR targets 1.4

if we need to revert any behavior changes first, we would probably want to backport, i.e. target 1.3.3

**Comment by kurchi1205** [comment]
ok thanks

**Comment by kurchi1205** [comment]
I have maintained the original behaviour and issued a warning in case all the columns are non numeric

**Comment by simonjayhawkins** [comment]
> I have maintained the original behaviour and issued a warning in case all the columns are non numeric

If this is the agreed way forward, we need two PRs.

one restoring the old behavior targeting 1.3.3 and one adding the deprecation to target 1.4

can you add release notes and also we need to test for the deprecation warning

**Comment by simonjayhawkins** [comment]
sorry for appearing to contradict the earlier comment, but at that point the suggestion was to issue a warning and keep the current behavior.

if done, a PR restoring 1.2 behavior should be done first. adding a deprecation warning should probably wait till that's agreed and merged. There is a risk of conflicts making the backport less straightforward if done in the opposite order.

**Comment by kurchi1205** [comment]
Ok so right now , I will create a pull request that restores 1.2 behaviour ie. numeric_only is set to True  . After that once the next steps are decided  , I will add the deprecation warning in another pull request

**Comment by simonjayhawkins** [comment]
> Ok so right now , I will create a pull request that restores 1.2 behaviour ie. numeric_only is set to True .

I'm not so sure that numeric_only is set to True, we need to identify when the user specifies `numeric_only` explicitly?

my understanding is that to fix the regression, we need to restore the default behavior when the Dataframe or Groupby object has only one column and that column is not numeric but also drop the column from the results when `numeric_only=True` is explicitly passed. That should probably not need any changes to existing tests and with the code samples from the two issues added and passing?

That PR (or this one if pushing changes here) would target 1.3.3 and need a release note in doc/source/whatsnew/v1.3.3.rst

That would effectively close the existing issue from the regression perspective but we would want to also confirm what the expected future behavior would be. We could open a new issue or keep that one open for further discussion.

cc @Dr-Irv @jbrockmendel @rhshadrach

**Comment by Dr-Irv** [comment]
> my understanding is that to fix the regression, we need to restore the default behavior when the Dataframe or Groupby object has only one column and that column is not numeric but also drop the column from the results when `numeric_only=True` is explicitly passed.

It's not just for _one_ column.  If _all_ of the columns are non-numeric, then the reduction (e.g., `sum`) is applied and all columns are returned.   Example below is with pandas 1.2.5:

```python
>>> df = pd.DataFrame({"x": [1,0,1,1,0], "y": [pd.Timedelta(i, "days") for i in range(1,6)], "z": [pd.Timedelta(i*10, "days") for i in range(1,6)]})
>>> df
   x      y       z
0  1 1 days 10 days
1  0 2 days 20 days
2  1 3 days 30 days
3  1 4 days 40 days
4  0 5 days 50 days
>>> df.groupby("x").sum()
       y       z
x
0 7 days 70 days
1 8 days 80 days
```
Interestingly, if `numeric_only=True`, you get the same result, which one could argue is a bug
```python
>>> df.groupby("x").sum(numeric_only=True)
       y       z
x
0 7 days 70 days
1 8 days 80 days
```
> That would effectively close the existing issue from the regression perspective but we would want to also confirm what the expected future behavior would be. We could open a new issue or keep that one open for further discussion.

I agree with this.

**Comment by kurchi1205** [comment]
```      
def _resolve_numeric_only(self, numeric_only: bool | lib.NoDefault) -> bool:
        """
        Determine subclass-specific default value for 'numeric_only'.

        For SeriesGroupBy we want the default to be False (to match Series behavior).
        For DataFrameGroupBy we want it to be True (for backwards-compat).

        Parameters
        ----------
        numeric_only : bool or lib.no_default

        Returns
        -------
        bool
        """
        # GH#41291
    
        if numeric_only is lib.no_default:
            # i.e. not explicitly passed by user
            if self.obj.ndim == 2:
                # i.e. DataFrameGroupBy
                numeric_cols = [x for x in self.obj.columns if is_numeric_dtype(self.obj[x]) ]
                num_cols_to_agg = len(set(numeric_cols) - set(self.grouper.names))
                if num_cols_to_agg > 0:
                    numeric_only = True
                else:
                    numeric_only = None
            else:
                numeric_only = None

        # error: Incompatible return value type (got "Union[bool, NoDefault]",
        # expected "bool")
        return numeric_only  # type: ignore[return-value]
```

For #43108 , if numeric_only is set to True explicitly then column v will be dropped otherwise , 'v' will be aggregated.
for #42395, if numeric_only is set to True explicitly then column value will be dropped otherwise,it will be aggregated .

for https://github.com/pandas-dev/pandas/pull/43154#issuecomment-905450396  ,  for default value of numeric_only , it will aggregate and explicitly for numeric_only=True , it will drop the columns 
I think this is the behaviour expected

**Comment by Dr-Irv** [comment]
> numeric_only = None
>             else:
>                 numeric_only = None

Should be `False`, not `None`.  If you're going to propose these changes, please make sure to create tests that test various combinations of numeric and non numeric columns, with one, two and three columns.

**Comment by kurchi1205** [comment]
yeah , if this is agreed by everyone , i will definitely create tests in the pull request .

**Comment by kurchi1205** [comment]
> > So as this is possible, even though it is making this more complicated than necessary for now, the regression should probably be fixed.
> 
> Sure. So IIUC the idea is to change _resolve_numeric_only to look like:
> 
> ```
> def _resolve_numeric_only(self, numeric_only) -> bool:
>         if numeric_only is lib.no_default:
>             # i.e. not explicitly passed by user
>             if self.obj.ndim == 2:
>                 # i.e. DataFrameGroupBy
>                 numeric_only = True
>                 obj = self._obj_with_exclusions   # or self.obj?  need to be careful here
>                 check = obj._get_numeric_data()
>                 if len(obj.columns) and not len(check.columns):
>                     warnings.warn("... Explicitly pass numeric_only ...")
>                     numeric_only=False
>                      
>             else:
>                 numeric_only = False
>     
> ```

So are we changing `_resolve_numeric_only ` or restoring 1.2 behaviour ?

**Comment by Dr-Irv** [comment]
> So are we changing `_resolve_numeric_only ` or restoring 1.2 behaviour ?

I believe that the proposed change to `_resolve_numeric_only ` will restore the 1.2 behavior.  If you can make this PR do that, and add appropriate tests, and make sure that all current tests pass, we should be in good shape.

**Comment by kurchi1205** [comment]
Can someone help l me to solve timeout issues in Azure pipelines?

**Comment by Dr-Irv** [comment]
> Can someone help l me to solve timeout issues in Azure pipelines?

Before doing that, can you add tests corresponding to #42395, as well as a test with 2 columns being non-numeric, and a test with 1 numeric and 1 non-numeric column.

Also, please include a comment with a pointer to the issues #42395 and #43108 in both the code you fixed and the tests.

**Comment by kurchi1205** [comment]
@Dr-Irv yeah I have added now .

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (rhshadrach) [inline pandas/tests/groupby/aggregate/test_aggregate.py:160]: Nit: only the `result = g.sum()` should be in the context manager; other lines should be before/after it as this is where the warning is supposed to be generated.

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (rhshadrach) [inline pandas/tests/groupby/aggregate/test_aggregate.py:120]: Move all tests to test_function, there is a class there TestNumericOnly. test_aggregate is meant for tests on e.g. DataFrameGroupBy.aggregate.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR! Some CI failures that need to be resolved. Also, the warning message will need to be fleshed out (@jbrockmendel used ellipses to indicate that it needs filled in with more details), however, I think there is an outstanding question as to whether there should be a warning for 1.3.x (ref: https://github.com/pandas-dev/pandas/issues/42395#issuecomment-908690066). I would hold off improving it until that is resolved.

Other than this, just some minor requests about the tests, but otherwise looks good!

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (jbrockmendel) [inline pandas/tests/groupby/aggregate/test_aggregate.py:134]: nitpick: can you call this `gb` instead of `g`?  1-letter variable names make things harder to grep for

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (jbrockmendel) [inline pandas/tests/groupby/aggregate/test_aggregate.py:136]: try to put the minimum amount of code inside the tm.assert_produces_warning context.  I think just the .sum() call produces the warning, so the other two lines would go outside

File: pandas/core/groupby/groupby.py
Comment (rhshadrach) [inline pandas/core/groupby/groupby.py:1123]: We have alignment on removing this warning - can you do so? A warning will be added in the deprecation of this behavior; that is to be targeted for 1.4 so it is not to be done here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/aggregate/test_aggregate.py
Comment (kurchi1205) [reply pandas/tests/groupby/aggregate/test_aggregate.py:136]: yeah I am making all the necessary changes

File: pandas/core/groupby/groupby.py
Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1123]: ok so no warning will be added for 1.3.x , only a `FutureWarning` in 1.4

File: pandas/core/groupby/groupby.py
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py:1105]: can you restore this whitespace. docstrings should start with a one-liner summary distinct from the expanded description.

File: pandas/core/groupby/groupby.py
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py:1108]: restore this one too.

File: pandas/core/groupby/groupby.py
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py:1112]: and this

File: pandas/core/groupby/groupby.py
Comment (simonjayhawkins) [inline pandas/core/groupby/groupby.py:1128]: can you restore this too. The mypy error needs to be ignored as mypy does not know that `lib.no_default` is a singleton.

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@kurchi1205 can you add a release note.

File: pandas/core/groupby/groupby.py
Comment (jreback) [inline pandas/core/groupby/groupby.py:1120]: put in a comment in reference to this issue and explain *why* we are doing this (compatiblity)

File: pandas/tests/groupby/test_function.py
Comment (jreback) [inline pandas/tests/groupby/test_function.py:267]: move near this tests: 

pandas/tests/groupby/test_groupby.py:def test_groupby_empty_multi_column(as_index, numeric_only):


OT we should move the tests of numeric_only=False to a separate location

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/groupby.py
Comment (kurchi1205) [reply pandas/core/groupby/groupby.py:1128]: > @kurchi1205 can you add a release note.

I assume the release note should be added in v1.4.0.rst

File: pandas/core/groupby/groupby.py
Comment (jbrockmendel) [inline pandas/core/groupby/groupby.py:1123]: and in a follow-up PR a warning goes here?

**Comment by Dr-Irv** [comment]
> @kurchi1205 can you add a release note.

I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py]: I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py]: I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py]: I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py]: > can you restore this too. The mypy error needs to be ignored as mypy does not know that lib.no_default is a singleton.
I"ve done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py]: > > @kurchi1205 can you add a release note.
> 
> I assume the release note should be added in v1.4.0.rst

We discussed at pandas dev meeting today, and put this in v1.3.3.rst

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py:1175]: > put in a comment in reference to this issue and explain _why_ we are doing this (compatiblity)

I've done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/tests/groupby/test_function.py
Comment (Dr-Irv) [reply pandas/tests/groupby/test_function.py]: > move near this tests:
> 
> pandas/tests/groupby/test_groupby.py:def test_groupby_empty_multi_column(as_index, numeric_only):
> 
I've done this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

File: pandas/core/groupby/groupby.py
Comment (Dr-Irv) [reply pandas/core/groupby/groupby.py:1178]: > and in a follow-up PR a warning goes here?

I put a `# TODO` for this in https://github.com/pandas-dev/pandas/pull/43154/commits/45f54d6cb548828072fc2d5885687d4d4db5b712

**Comment by Dr-Irv** [comment]
@jbrockmendel @simonjayhawkins @jreback  Based on discussion at today's meeting, I updated the PR based on previous comments.  Please review to see if it's good to go for 1.3.3.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. in theory this might have a small penalty, but its ok as we need the correct results.

**Comment by Dr-Irv** [comment]
> lgtm. in theory this might have a small penalty, but its ok as we need the correct results.

There's a failure https://github.com/pandas-dev/pandas/pull/43154/checks?check_run_id=3551554045#step:7:70 that needs to be investigated before we can merge.

**Comment by Dr-Irv** [comment]
> 
> There's a failure https://github.com/pandas-dev/pandas/pull/43154/checks?check_run_id=3551554045#step:7:70 that needs to be investigated before we can merge.

I put something in to address this failure related to looking for the `FutureWarning`.  See https://github.com/pandas-dev/pandas/pull/43154/commits/d46091edc35be4606696431a8dd80ab25d335630

**Comment by Dr-Irv** [comment]
@jreback I think this is good to go.  Two CI failures.  One is an issue in building the docs due to a connection failure.  The other is the Windows timeout.

**Review by jreback — APPROVED** [review]
lgtm thanks @kurchi1205 and @Dr-Irv

**Comment by jreback** [comment]
@simonjayhawkins IIRC there were some duplicates of the original issue that we can now see if they are solved? can you link

**Comment by Dr-Irv** [comment]
> @simonjayhawkins IIRC there were some duplicates of the original issue that we can now see if they are solved? can you link

There was the original issue #42395 and a "duplicate" that I created #43108.  @kurchi1205 added tests from both of those.  Not aware of any others.

**Comment by simonjayhawkins** [comment]
> > @simonjayhawkins IIRC there were some duplicates of the original issue that we can now see if they are solved? can you link
> 
> There was the original issue #42395 and a "duplicate" that I created #43108. @kurchi1205 added tests from both of those. Not aware of any others.

There was also #43209 that was a regression from the same commit #41706, but it doesn't look like the change in this PR has fixed that one.

**Comment by simonjayhawkins** [comment]
> this test was added in #41706

sorry this was changed... from https://github.com/pandas-dev/pandas/pull/43154#issuecomment-906361576

> (there was also a change to test_cython_agg_nothing_to_agg which i'm not sure about. It looks like a case that used to raise, now returns an empty DataFrame, not checked to see if that is a bugfix or an api change, i'm considering that one out of scope for now)

I'll open an new issue rather than continue the discussion here.

**Comment by simonjayhawkins** [comment]
> I'll open an new issue rather than continue the discussion here.

#43501

**Pull request merged**
